src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/TelemetryInitializerBase.cs

```diff
@@ -33,15 +33,18 @@ public void Initialize(ITelemetry telemetry)
                 return;
             }
 
-            var request = context.Features.Get<RequestTelemetry>();
-
-            if (request == null)
+            lock (context)
```

Comment:
this is a partial solution and it will not work when customer will try to access `RequestTelemetry` himself. As we discussed in the issue - the best solution will be to move `RequestTelemetry` from features back to DI and make sure it has most of the properties initialized as early as possible.

Even the first step will make the possibility of a crash way less possible.